import pandas as pd
import psycopg2
from psycopg2 import OperationalError
import os
from dotenv import load_dotenv

load_dotenv()

def load_data():
    try:
        conn = psycopg2.connect(
            host=os.getenv("DB_HOST"),
            dbname=os.getenv("DB_NAME"),
            user=os.getenv("DB_USER"),
            password=os.getenv("DB_PASSWORD"),
            port=int(os.getenv("DB_PORT", 5432))
        )
        query = "SELECT * FROM {{ config.source_details.table_name }} LIMIT 1000"
        df = pd.read_sql(query, conn)
        return df
    except OperationalError as e:
        raise Exception(f"PostgreSQL connection failed: {str(e)}")
    finally:
        if 'conn' in locals():
            conn.close()
